frontend/lib/resolution/scope-queries.cpp

```diff
@@ -2265,9 +2308,13 @@ doResolveUseStmt(Context* context, const Use* use,
     }
 
     const Scope* foundScope = findUseImportTarget(context, scope, r,
-                                                  expr, VIS_USE, oldName);
+                                                  expr, VIS_USE, oldName,
+                                                  namedModulesSet);
     if (foundScope != nullptr) {
 
+      // note that this module is named in 'use'
+      addNamedModule(r, namedModulesSet, foundScope->id());
```

Comment:
I would've thought that the call to `findUseImportTarget` above would've already gotten this. Though looking back, I see that it doesn't call `addNamedModule` for `super` and `this`, while this would catch that. Why the fragmentation?

Reply:
We don't want to get `super` or `this` anyway. Turns out this line isn't necessary (earlier iterations of the code needed it). I've removed it.